### ISSUE TITLE
Using sergiogasquez/esp-rs-env:esp32 with Github workflow

### ISSUE THREAD

**Issue by callmephilip** [SergioGasquez/esp-rs-container#4]
@SergioGasquez have you had any luck using `sergiogasquez/esp-rs-env:esp32` container in GH workflows? I am currently trying to get it to work [here](https://github.com/bakery/rust-esp32-std-demo/blob/feature/ota-updates/.github/workflows/ci.yml#L17), however the flow breaks on the [build command](https://github.com/bakery/rust-esp32-std-demo/blob/feature/ota-updates/.github/workflows/ci.yml#L48) with the following message:

`error: toolchain 'esp' is not installed`

I ran some diagnostics as well with ` rustup toolchain list` and i am getting: `no installed toolchains`

**Comment by SergioGasquez** [comment]
Hello! I haven't tried to use the container in a GH action so far.  I see that you are using [`--user root`](https://github.com/bakery/rust-esp32-std-demo/blob/feature/ota-updates/.github/workflows/ci.yml#L19) could you try using `--user esp`?

Let me know how it goes!

**Comment by callmephilip** [comment]
with `--user esp`, Checkout Action fails with

```
Run actions/checkout@v3
  with:
    set-safe-directory: false
    repository: bakery/rust-esp3[2](https://github.com/bakery/rust-esp32-std-demo/runs/6385350078?check_suite_focus=true#step:3:2)-std-demo
    token: ***
    ssh-strict: true
    persist-credentials: true
    clean: true
    fetch-depth: 1
    lfs: false
    submodules: false
  env:
    cargo_file_path: ./Cargo.toml
    wifi_ssid: ***
    wifi_password: ***
/usr/bin/docker exec  9d9e25[3](https://github.com/bakery/rust-esp32-std-demo/runs/6385350078?check_suite_focus=true#step:3:3)[4](https://github.com/bakery/rust-esp32-std-demo/runs/6385350078?check_suite_focus=true#step:3:4)f914d14eab24e876b1ea149416cd0034fed9a0e8a434902d7[5](https://github.com/bakery/rust-esp32-std-demo/runs/6385350078?check_suite_focus=true#step:3:5)[7](https://github.com/bakery/rust-esp32-std-demo/runs/6385350078?check_suite_focus=true#step:3:7)045b4 sh -c "cat /etc/*release | grep ^ID"
Syncing repository: bakery/rust-esp32-std-demo
Getting Git version info
Deleting the contents of '/__w/rust-esp32-std-demo/rust-esp32-std-demo'
Initializing the repository
  /usr/bin/git init /__w/rust-esp32-std-demo/rust-esp32-std-demo
  /__w/rust-esp32-std-demo/rust-esp32-std-demo/.git: Permission denied
  Error: The process '/usr/bin/git' failed with exit code 1
```

**Comment by SergioGasquez** [comment]
Hello!

It looks like it's trying to checkout in `/__w/rust-esp32-std-demo/rust-esp32-std-demo/` where user `esp` has no permissions, we should try to work in `/home/esp` or something similar. I see there is a [$GITHUB_WORKSPACE](https://docs.github.com/es/actions/learn-github-actions/environment-variables#default-environment-variables) environment variable but don't really know if we can edit it, and if it also applies when using containers. 
I will update you if I find a solution!

**Comment by callmephilip** [comment]
@SergioGasquez ok. this makes sense. i don't think we can tweak the location where the code gets checked out to. one other thing i have noticed is if i try to run using `esp` user (no checkout, just see if the toolchain is there)

```
container: 
      image: sergiogasquez/esp-rs-env:esp32
      options: --user esp
steps:
      - name: Debug | Toolchain
        run: rustup toolchain list 
```

This will fail with access permission error as well since rustup will look inside $HOME which is set to `~/home/github` by the CI server. I am wondering if [this](https://github.com/SergioGasquez/esp-rs-container/blob/main/xtensa-base.Dockerfile#L31) `.bashrc` update propagates correctly

**Comment by callmephilip** [comment]
Found some clarification here: https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#user

It looks like GH discourages using `USER` directive in Docker files

**Comment by callmephilip** [comment]
I changed the approach a little bit by moving things into a separate action, as you can see [here](https://github.com/bakery/rust-esp32-std-demo/blob/feature/ota-updates/.github/actions/esp-build/Dockerfile). Had to tweak some stuff by wrapping another Docker file around your image. Got it to run check out and recognize toolchains installed. However building it now triggers a cryptic looking error. Logs below

**Comment by callmephilip** [comment]
[espbuilderror.logs.txt.zip](https://github.com/SergioGasquez/esp-rs-container/files/8688652/espbuilderror.logs.txt.zip)

**Comment by SergioGasquez** [comment]
Hello @callmephilip! Great workaround! The error that you were seeing is because the esp-idf environment was not activated for the user root, as you can see on the log it was downloading esp-idf again under [`.embuild`](https://github.com/bakery/rust-esp32-std-demo/runs/6424338097?check_suite_focus=true#step:3:18254) directory instead of using the one installed in the image under /home/esp/.espressif/frameworks/esp-idf.v4.4. I created a PR to merge the changes into your repo: bakery/rust-esp32-std-demo/pull/2

Be aware that the images are hosted under my personal account and they are changing rapidly, we plan on moving them under the espressif's Dockerhub account in the upcoming weeks. For example I just changed the [name of the folder where esp-idf is installed](https://github.com/SergioGasquez/esp-rs-container/commit/9bd3cd470e413c018643deca67b9485c3a7c714c), so once this [action finishes](https://github.com/SergioGasquez/rust-esp32-std-demo/actions/runs/2320243286),  we will need to [change this line](https://github.com/SergioGasquez/rust-esp32-std-demo/blob/01b535d15136960853f2803cfac598112febf631/.github/actions/esp-build/entrypoint.sh#L6).

Thanks for your contribution!

**Comment by callmephilip** [comment]
Good stuff @SergioGasquez ! Thank you for your help.

**Comment by callmephilip** [comment]
@SergioGasquez any idea why this is failing: https://github.com/bakery/rust-esp32-std-demo/blob/feature/ota-updates/.github/actions/esp-build/entrypoint.sh#L9 with

```
/home/esp/.cargo/bin/cargo-espflash: 1: Not: not found
```

`cargo-espflash` is definitely installed and the script is there

**Comment by callmephilip** [comment]
not sure what was going on tbh. ended up reinstalling cargo-espflash and it seems to be working now: https://github.com/bakery/rust-esp32-std-demo/blob/feature/ota-updates/.github/actions/esp-build/entrypoint.sh#L13

**Comment by SergioGasquez** [comment]
The issue might be related to the change of release artifact name. This should be fixed in our 1.61 release, which should be out soon (late this week or next week).

Reinstalling cargo-espflash is a good solution for the moment as it should be something only temporary-

Thanks for all your contributions!

**Comment by callmephilip** [comment]
Moved build action for esp32 to its own repo so people can use it - https://github.com/bakery/actions-esp-rs-build
